incubator/java-spring-boot2/image/project/Dockerfile

```diff
@@ -1,21 +1,12 @@
-FROM maven:3.6-ibmjava-8 as compile
-LABEL maintainer="IBM Java Engineering at IBM Cloud"
-
-# Ensure up to date / patched OS
-RUN  apt-get -qq update \
-  && apt-get -qq install -y curl wget xmlstarlet unzip \
-  && DEBIAN_FRONTEND=noninteractive apt-get -qq upgrade -y \
-  && apt-get -qq clean \
-  && rm -rf /tmp/* /var/lib/apt/lists/* \
-  && mkdir -p /mvn/repository
+FROM {{APPSODY_STACK}} as compile
```

Comment:
So I assume this is attempting to replace the installation of the latest OS updates with simply using those from the stack image itself? While this seems a sensible thing, we should probably make sure we are all OK with as a general policy (to date, I think we have always stated that the two builds are independent). As an aside, I assume this change is not actually required for the template change...might it be better to do this in two separate PRs?

Reply:
This change cascades from the template change, as it is using variable substitution. The APPSODY_STACK is a variable in the Dockerfile that is cascaded down. I could pull it out, but I'd rather not. I think the behavior change all goes together.

As for updates: this is all about how something is _built_ .. not how it is run. The above changes the first stage (as compile), not the second stage (the run image). This means that the same stack will be used when building locally, and when building in the pipeline, which seems much more predictable and consistent to me.

Reply:
I tend to agree - keeping the stack the same is also good for ensuring you don't inadvertently introduce changes as you move from run->build->deploy